### PR TITLE
Split box.stat page by functions

### LINKED ISSUE #2862: [2pt] Restructure box.stat page
**Product:** Tarantool
**Root document:** https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_stat/
**SME:** TBD

## Details

Currently, there's a single landing-like page for the `box.stat` submodule reference. All the submodule functions with all the details are describe in a flat list which is quite a mess and really not convenient to read and use. 

Needs to: 
* Create a separate page for each function.
* On the [main page](https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_stat/), there should be just an introduction text and ToC navigating to the pages of the functions.

### PR DESCRIPTION
**Pull request by p7nov**
Fixes #2862

### CODE REVIEW

File: doc/reference/reference_lua/box_stat/net.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
+==============================
```
Comment (NickVolynkin) [inline doc/reference/reference_lua/box_stat/net.rst]: Wrong heading syntax. Please use single line of `=` under the text. https://www.tarantool.io/en/doc/latest/contributing/docs/markup/headings/

Comment (p7nov) [reply doc/reference/reference_lua/box_stat/net.rst]: Thanks, fixed

File: doc/reference/reference_lua/box_stat.rst [hunk 2]
```diff
@@ -6,273 +6,36 @@ Submodule box.stat
...
 
-.. _box_introspection-box_stat_reset:
+        *   - Name
+            - Use
 
-Use ``box.stat.reset()`` to reset the statistics of ``box.stat()``,
-``box.stat.net()``, ``box.stat.vinyl()`` and
-:ref:`box.space.index <box_space-space_index>`.
+        *   - :doc:`./box_stat/stat`
+            - Shows request statistics
```
Comment (patiencedaur) [inline doc/reference/reference_lua/box_stat.rst]: Consider using the infinitive: "Show request statistics". For reference, check similar pages on the same level.

File: doc/reference/reference_lua/box_stat.rst [hunk 3]
```diff
@@ -6,273 +6,36 @@ Submodule box.stat
...
 
-In the tables that ``box.stat()`` returns:
+        *   - :doc:`./box_stat/net`
+            - Shows network activity
 
-* ``total`` stands for "total number of requests processed per second since
-  the server started",
-* ``rps`` stands for "average number of requests per second in the last 5 seconds".
+        *   - :doc:`./box_stat/vynil`
```
Comment (patiencedaur) [inline doc/reference/reference_lua/box_stat.rst]: The name of the storage engine is vinyl, not vynil. Please re-check this across the whole PR, including document names.

File: doc/reference/reference_lua/box_stat/net.rst [hunk 4]
```diff
@@ -0,0 +1,45 @@
+box.stat.net()
+==============================
```
Comment (patiencedaur) [inline doc/reference/reference_lua/box_stat/net.rst]:
```suggestion
box.stat.net()
==============
```

The underline must be the same length as the header. Check our [markup guide](https://www.tarantool.io/en/doc/latest/contributing/docs/markup/headings/).

File: doc/reference/reference_lua/box_stat/reset.rst [hunk 5]
```diff
@@ -0,0 +1,12 @@
+.. _box_introspection-box_stat_reset:
+
+box.stat.reset()
+==============================
```
Comment (patiencedaur) [inline doc/reference/reference_lua/box_stat/reset.rst]:
```suggestion
box.stat.reset()
================
```

File: doc/reference/reference_lua/box_stat/stat.rst [hunk 6]
```diff
@@ -0,0 +1,64 @@
+box.stat()
+==============================
```
Comment (patiencedaur) [inline doc/reference/reference_lua/box_stat/stat.rst]:
```suggestion
box.stat()
==========
```

File: doc/reference/reference_lua/box_stat/vynil.rst [hunk 7]
```diff
@@ -0,0 +1,187 @@
+.. _box_introspection-box_stat_vinyl:
+
+box.stat.vynil()
+==============================
+
```
Comment (patiencedaur) [inline doc/reference/reference_lua/box_stat/vynil.rst]:
```suggestion
box.stat.vinyl()
================

```

File: doc/reference/reference_lua/box_stat/vynil.rst [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+.. module:: box.stat
+
+.. function:: vynil()
```
Comment (patiencedaur) [inline doc/reference/reference_lua/box_stat/vynil.rst]:
```suggestion
.. function:: vinyl()
```

**Review by patiencedaur — CHANGES_REQUESTED** [review]

File: doc/reference/reference_lua/box_stat.rst (hunk 3 above)
Comment (p7nov) [reply doc/reference/reference_lua/box_stat.rst]: Wow, my bad, thanks for spotting this. Will fix this all around.

**Review by patiencedaur — COMMENTED** [review]
One more thing. The box.stat submodule link is now at the top of the menu, which violates the alphabetical order. Please move it to the right place. (Check the `box.rst` file.)
image

**Review by patiencedaur — APPROVED** [review]

**Pull request merged**